lib/pyfrc/physics/tankmodel.py

```diff
@@ -0,0 +1,233 @@
+                       l_kv: float, l_ka: float, l_vi: float,
+                       r_kv: float, r_ka: float, r_vi: float):
+        '''
+            Use the constructor if you have measured ``kv``, ``ka``, and
+            ``Vintercept`` for your robot. Use the :func:`.theory` function
+            if you haven't.
+            
+            ``Vintercept`` is the minimum voltage required to generate enough
+            torque to overcome steady-state friction (see the paper for more
+            details).
+        
+            :param motor_config: Motor specification
+            :param x_wheelbase: Wheelbase of the robot (in feet)
+            :param robot_mass: Mass of robot (in lb)
+            :param l_kv: Left side ``kv``
+            :param l_ka: Left side ``ka``
+            :param l_vi: Left side ``Vintercept``
+            :param r_kv: Right side ``kv``
+            :param r_ka: Right side ``ka``
+            :param r_vi: Right side ``Vintercept``
+        '''
+        
+        x_wheelbase *= FT_TO_M
+        robot_mass *= LB_TO_KG
+        
+        self._lmotor = MotorModel(motor_config, l_kv, l_ka, l_vi)
+        self._rmotor = MotorModel(motor_config, r_kv, r_ka, r_vi)
+        self._x_wheelbase = x_wheelbase
+        
+        self.l_position = 0
+        self.r_position = 0
+        
+        # this uses the formula for the moment of inertial for a circle...
+        # which seems to give good feeling results for me. If you find it
+        # doesn't work, compute your own and set it :)
+        
+        #: The model computes a moment of inertia for your robot based on the
+        #: given mass and wheelbase If you wish to use a different moment of
+        #: inertia, set this property after constructing the object
+        self.inertia = .5 * robot_mass * math.pow(self._x_wheelbase/2.0, 2)
+    
+    def get_vector(self, l_motor: float, r_motor: float, tm_diff: float) -> typing.Tuple[float, float]:
+        '''
+            Given motor values, retrieves the vector of (velocity, rotation) for your robot
+        
+            :param l_motor:    Left motor value (-1 to 1); -1 is forward
+            :param r_motor:    Right motor value (-1 to 1); 1 is forward
+            :param tm_diff:    Elapsed time since last call to this function
+
+            :returns: velocity of robot (ft/s), clockwise rotation of robot (radians/s)
+            
+            .. note:: If you are using more than 2 motors, it is assumed that
+                      all motors on each side are set to the same speed. Only
+                      pass in one of the values from each side
+        '''
+        
+        l = self._lmotor.compute(-l_motor, tm_diff)
+        r = self._rmotor.compute(r_motor, tm_diff)
+        
+        # Tank drive motion equations
+        velocity = (l + r) * 0.5
+        rotation = (l - r) / float(self._x_wheelbase)
+        
+        velocity *= M_TO_FEET
+        rotation *= self.inertia
```

Comment:
Multiplying by moment of inertia here makes this function return angular momentum, not angular velocity. Angular momentum `L = J * omega`.

Reply:
By doing conservation of angular momentum (sum of angular momenta = J*omega), I get the following. Let `b` be the robot's wheelbase over 2, `m` be the mass, `l` and `r` be the velocities of each side of the robot, `J` be the moment of inertia, and `omega` be the angular velocity of the robot.
```
-b * m * l + b * m * r = J * omega
b * m * (r - l) = J * omega
omega = b * m * (r - l) / J
```

Reply:
That equation makes the robot go crazy :(

Reply:
Converting to this equation is more realistic, and switching to a rectangular moment of inertia fixed the craziness issues.